### DISCUSSION TITLE
编译到 wasm 但不生成 html

### DISCUSSION THREAD

**Discussion by guyutongxue** [xmake-io/xmake#3440] [General]
貌似 xmake 将平台设置到 wasm / emscripten 后，会设置输出选项为 `-o $(name).html`

https://github.com/xmake-io/xmake/blob/aeb7ae95327b68b3d5d814ae6308940397e6394e/xmake/platforms/wasm/xmake.lua#L37

但是大部分时候我们只需要 js 胶水 + wasm，也就是输出选项是 `-o $(name).js` [参考](https://emscripten.org/docs/tools_reference/emcc.html#emcc-o-target)

这个能不能自定义呢？

**Comment by waruqi** [comment]
生成 html 的同时，也会同时生成 .js 和 wasm ，既然你只用这两，就假装没看见 .html 不就好了。

或者自己 after_build 里面，把 .html 自己删了。。

暂时不能自定义，也没必要

**Reply by guyutongxue** [reply]
你说得有道理，确实没有足够的必要性——我只是犯了一些强迫症而已

**Comment by myzhan** [comment]
还是有一点影响，例如 emscripten 的这个连接选项 -s EXPORT_NAME=initWasmModule，在 -o xxx.html 和 -o xxx.js 的时候，有不同的行为。

https://github.com/emscripten-core/emscripten/issues/10086
